rust/sdk-processor/src/steps/stake_processor/stake_storer.rs

```diff
@@ -0,0 +1,190 @@
+use crate::{
+    processors::stake_processor::StakeProcessorConfig,
+    utils::database::{execute_in_chunks, get_config_table_chunk_size, ArcDbPool},
+};
+use ahash::AHashMap;
+use anyhow::Result;
+use aptos_indexer_processor_sdk::{
+    traits::{async_step::AsyncRunType, AsyncStep, NamedStep, Processable},
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use async_trait::async_trait;
+use processor::{
+    db::common::models::stake_models::{
+        current_delegated_voter::CurrentDelegatedVoter,
+        delegator_activities::DelegatedStakingActivity,
+        delegator_balances::{CurrentDelegatorBalance, DelegatorBalance},
+        delegator_pools::{CurrentDelegatorPoolBalance, DelegatorPool, DelegatorPoolBalance},
+        proposal_votes::ProposalVote,
+        staking_pool_voter::CurrentStakingPoolVoter,
+    },
+    processors::stake_processor::{
+        insert_current_delegated_voter_query, insert_current_delegator_balances_query,
+        insert_current_delegator_pool_balances_query, insert_current_stake_pool_voter_query,
+        insert_delegator_activities_query, insert_delegator_balances_query,
+        insert_delegator_pool_balances_query, insert_delegator_pools_query,
+        insert_proposal_votes_query,
+    },
+};
+
+pub struct StakeStorer
+where
+    Self: Sized + Send + 'static,
+{
+    conn_pool: ArcDbPool,
+    processor_config: StakeProcessorConfig,
+}
+
+impl StakeStorer {
+    pub fn new(conn_pool: ArcDbPool, processor_config: StakeProcessorConfig) -> Self {
+        Self {
+            conn_pool,
+            processor_config,
+        }
+    }
+}
+
+#[async_trait]
+impl Processable for StakeStorer {
+    type Input = (
+        Vec<CurrentStakingPoolVoter>,
+        Vec<ProposalVote>,
+        Vec<DelegatedStakingActivity>,
+        Vec<DelegatorBalance>,
+        Vec<CurrentDelegatorBalance>,
+        Vec<DelegatorPool>,
+        Vec<DelegatorPoolBalance>,
+        Vec<CurrentDelegatorPoolBalance>,
+        Vec<CurrentDelegatedVoter>,
+    );
+    type Output = ();
+    type RunType = AsyncRunType;
+
+    async fn process(
+        &mut self,
+        input: TransactionContext<(
+            Vec<CurrentStakingPoolVoter>,
+            Vec<ProposalVote>,
+            Vec<DelegatedStakingActivity>,
+            Vec<DelegatorBalance>,
+            Vec<CurrentDelegatorBalance>,
+            Vec<DelegatorPool>,
+            Vec<DelegatorPoolBalance>,
+            Vec<CurrentDelegatorPoolBalance>,
+            Vec<CurrentDelegatedVoter>,
+        )>,
+    ) -> Result<Option<TransactionContext<Self::Output>>, ProcessorError> {
+        let per_table_chunk_sizes: AHashMap<String, usize> = self
+            .processor_config
+            .default_config
+            .per_table_chunk_sizes
+            .clone();
```

Comment:
Now that I think about it, I think we should move this part to the processor so we avoid the clone every time process is called?